### DISCUSSION TITLE
Need help for writting slippage exploit of Tswap protocol

### DISCUSSION THREAD

**Discussion by BsacHunter** [Cyfrin/security-and-auditing-full-course-s23#182] [Q&A]
Can anybody tell me how to increase or decrease the current price of the tokens in the foundry test to check slippage.

Thanks

**Comment by EngrPips** [comment]
Hello @ashHunter319, If you are talking in the context of a Dex, then make a swap before the original swap, and that would already cause slippage.

Example:
A Pool has 20:5 tokenA:tokenB. If I want to swap 4 tokenA, then I am supposed to get 1 tokenB. But a guy beat me to it and swapped 4 token A for 1 token B. Now, the pool's ratio is 24:4 tokenA:tokenB. If I didn't protect myself with slippage and my trade still gets processed, then I will get less than 1 tokenB for my 4 tokenA because slippage already occurred before my transaction got processed.
